### ISSUE TITLE
New help rendering issues

### ISSUE THREAD

**Issue by ysndr** [pacak/bpaf#196]
I think #192 might wave introduced some regressions

consider

```rust
#[derive(bpaf::Bpaf, Clone)]
enum Commands {
    /// Multi
    /// Line
    /// Comment
    #[bpaf(command)]
    Lines {},
    #[bpaf(command)]
    Arguments(#[bpaf(external(arguments))] Arguments)
}


#[derive(bpaf::Bpaf, Clone)]
struct Arguments {
    #[bpaf(short('e'), argument("Arg"))]
    env: Vec<String>,

    #[bpaf(positional("POS"))]
    args: Vec<String>,
}
```

## multi line descriptions

```
$ prog --help

Usage: COMMAND ...							personnally, I would expect the argv0/program name to 
											show up there too but that's detail..
       
Available options:
    -h, --help  Prints help information

Available commands:
    lines      Multi Line Comment			< these used to be rendered as multiple lines before i'm 										
											  pretty sure
    arguments
```

```
$ prog lines --help
Multi										< Multiline...
Line
Comment


Available options:
    -h, --help  Prints help information
```

## argument errors

positionals work fine:

```
$ prog argumens
No such command or positional: `argumens`, did you mean `arguments`? 
```

options are a bit weird:

```
$ prog argumens -a erg
Expected an argument <POS>, got -a
					   |
                       shouldnt it first try to match options before goint into positionals?
                       also, POS isnot really expected, its optional (Vec) but more about that below		
```

## usage

```
$ prog argumens --help
Usage: -e Arg... <POS>...
        |		 L  [ <POS> ] ...
        |        or [ <POS> ... ]
        |
		L should be [ -e <Arg> ] ...
                or [ -e Arg ... ] 

--8<--
```

at least according to  http://docopt.org/, which unlike https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html at least contains subcommands and appears to be the same syntax git uses

**Comment by pacak** [comment]
> multi line descriptions

This bit is partially intentional and was changed earlier. Behavior picks bits of rust doc comments and multiline string literals.
When all the lines start at the same offset - it would try to flow it as a single line, to force a linebreak add a space like so:
```rust
#[derive(bpaf::Bpaf, Clone)]
enum Commands {
    /// Multi
    ///  Line
    ///  Comment
    #[bpaf(command)]
    Lines {},
    #[bpaf(command)]
    Arguments(#[bpaf(external(arguments))] Arguments)
}
```

This change is needed to generate better looking markdown and manpage documentation in semantic branch - both formats look better when they have a chance to fill the full line.

---
> argument errors

I'm working on something that would help to make errors like you describing

---
> usage

Great catch! I think it was wrong since the beginning, will fix.

**Comment by ysndr** [comment]
> This change is needed to generate better looking markdown and manpage documentation in semantic branch - both formats look better when they have a chance to fill the full line.

I see, so you're saying it's a me problem then 😄

> I'm working on something that would help to make errors like you describing

Great to hear, just wanted to bring it up because it was the first thing I stumbled over

**Comment by pacak** [comment]
> I see, so you're saying it's a me problem then 

Well... It's me problem deciding on a slightly strange behavior initially and then changing it without updating the documentation.

**Comment by pacak** [comment]
FWIW I released 0.8.0, it should help with your `--set key val` rendering in help.
